src/src/core/StringUtils.cpp

```diff
@@ -24,20 +24,37 @@ BEGIN_FTC_CORE
 namespace StringUtils
 {
 
+void ltrim(std::string &s)
+{
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](int ch) {
+        return !std::isspace(ch);
```

Comment:
`std::isspace(c)` is undefined for `c` not in range of `unsigned char`. Correct behavior is to cast `char` to `unsigned char`.

```suggestion
    s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](char ch) {
        return !std::isspace(static_cast<unsigned char>(ch));
```